frameworks/hdfs/alerts/HDFS_alert_rules.yml

```diff
@@ -0,0 +1,105 @@
+      labels:
+        severity: warning
+        service: hdfs
+      annotations:
+       summary: "Under replicated blocks on {{ $labels.instance }} for {{ $labels.service_name }} / {{ $labels.task_name }}"
+       description: "{{ $labels.instance }} / {{ $labels.service_name }} / {{ $labels.task_name }} has under replicated blocks equals to {{ $value }}"
+
+    - alert: NumStaleDataNodes
+      expr: sum by(service_name,task_name) (rate({__name__=~".*_dfs_FSNamesystem_StaleDataNodes"}[10m])) > 0
+      labels:
+        severity: warning
+        service: hdfs
+      annotations:
+       summary: "Number of stale DataNodes on {{ $labels.instance }} for {{ $labels.service_name }} / {{ $labels.task_name }}"
+       description: "{{ $labels.instance }} / {{ $labels.service_name }} / {{ $labels.task_name }} has number stale DataNodes equals to {{ $value }}"
+
+    - alert: MemHeapUsed
+      expr: (sum({__name__=~".*_jvm_JvmMetrics_MemHeapMaxM"}) / sum({__name__=~".*_jvm_JvmMetrics_MemHeapUsedM"})) < 0.8
+      for: 10m
+      labels:
+        severity: warning
+        service: hdfs
+      annotations:
+       summary: "Total service memory heap in used for {{ $labels.instance }}"
+       description: "{{ $labels.instance }} has heap in used greater then 0.8 (current value: {{ $value }})"
+
+    - alert: MemUsedPercent
+      expr: mem_used_percent > 80
+      for: 10m
+      labels:
+        severity: warning
+        service: hdfs
+      annotations:
+       summary: "Instance memory in used percent for {{ $labels.instance }}"
+       description: "{{ $labels.instance }} has memory in used greater then 80% (current value: {{ $value }}%)"
+
+  - name: hdfs_critical
+    rules:
+    - alert: InstanceDown
+      expr: up == 0
+      for: 5m
+      labels:
+        severity: critical
+        service: hdfs
+      annotations:
+        summary: "Instance {{ $labels.instance }} down"
+        description: "{{ $labels.instance }} of job {{ $labels.job }} has been down for more than 5 minutes."
+
+    - alert: NumInactiveSources
+      expr: (sum({__name__=~".*metricssystem_MetricsSystem_NumAllSources"}) - sum({__name__=~".*metricssystem_MetricsSystem_NumActiveSources"})) > 0
+      for: 5m
+      labels:
+        severity: critical
+        service: hdfs
+      annotations:
+       summary: "Number of inactive sources for {{ $labels.instance }}"
+       description: "{{ $labels.instance }} has number of inactive sources greater then 0 (current value: {{ $value }})"
+
+    - alert: VolumeFailures
+      expr: sum by (service_name,task_name) ({__name__=~".*_dfs_datanode_VolumeFailures"}) > 0
+      for: 5m
+      labels:
+        severity: critical
+        service: hdfs
+      annotations:
+       summary: "VolumeFailures on {{ $labels.instance }} for {{ $labels.service_name }} / {{ $labels.task_name }}"
+       description: "{{ $labels.instance }} / {{ $labels.service_name }} / {{ $labels.task_name }} has volume failure equals to {{ $value }}"
```

Comment:
```suggestion
       description: "Volume failures are occurring in the cluster. Please investigate as this can be a sign of faulty hardware that needs replacing."
```